### DISCUSSION TITLE
Unexpected error when handling authentication request to identity provider.

### DISCUSSION THREAD

**Discussion by fonteneu** [dcm4che/dcm4chee-arc-light#4599] [Q&A]
I followed the Docker Compose instructions on the wiki in great detail.

All docker services start with no errors in the logs.

When trying to do the initial login according to the OIDC instructions (visiting https://dcm4chee-arc:8843/admin/dcm4che/console), I get the following error in the login dialog:

**Unexpected error when handling authentication request to identity provider.**

The dmc4chee-arc hostname resolves to our IP address correctly.

I have found several other posts about this, but no solution:

https://groups.google.com/g/dcm4che/c/MEQ9XGZOKoY
https://groups.google.com/g/dcm4che/c/4vbrzF-67X8
https://github.com/keycloak/keycloak/issues/14438 (closed without explanation)

Anyone got ideas how to solve this?

**Comment by aniruddhaadak80** [comment]
Hey @fonteneu ,

This error often arises from configuration issues between the identity provider (Keycloak) and the dcm4chee-arc system. Here are some troubleshooting steps:

1. **Check Keycloak URL Configuration**: Ensure the `Identity Provider` URL in the dcm4chee-arc configuration points to the correct Keycloak endpoint, matching both protocol (`http` or `https`) and port.

2. **OIDC Client Configuration**: Verify that the Keycloak client settings for dcm4chee-arc allow for the redirect URI configured in Keycloak to match exactly with the dcm4chee-arc URI (e.g., `https://dcm4chee-arc:8843`).

3. **CORS & Proxy Issues**: Ensure CORS headers and any reverse proxies are properly configured, as these can sometimes block or misroute authentication responses. Check Docker network settings to confirm inter-service communication.

4. **Keycloak Logs**: Look into Keycloak logs for more specific error messages during the authentication request, as they may reveal missing configurations or permission issues.

These steps should help resolve common causes for this error when using Keycloak and dcm4chee-arc together.

Thank you.

**Reply by fonteneu** [reply]
Thanks for the quick reply. 

Regarding your first two points: I used the Docker Compose example on the wiki, can I assume that one is correct and up to date? 

Everything runs on the same server so no proxy and shouldn't be any cors issues. 

Nothing to be found in the keycloak logs.

**Reply by fonteneu** [reply]
Actually Docker shows no error for keycloak, but the raw logfile does have some errors related to database: https://pastebin.com/raw/QkGBHaTX

**Reply by fonteneu** [reply]
I reinstalled on a different Linux distribution and then it worked, though unclear why as I followed the same steps.
